core/integration/module_test.go

```diff
@@ -5396,19 +5431,37 @@ func daggerCallAt(modPath string, args ...string) dagger.WithContainerFunc {
 	}
 }
 
-func mountedPrivateRepoSocket(c *dagger.Client, t *testctx.T) (dagger.WithContainerFunc, func()) {
-	sockPath, cleanup := setupPrivateRepoSSHAgent(t)
+func privateRepoSetup(c *dagger.Client, t *testctx.T, tc vcsTestCase) (dagger.WithContainerFunc, func()) {
+	var socket *dagger.Socket
+	cleanup := func() {}
+	if tc.sshKey {
+		var sockPath string
+		sockPath, cleanup = setupPrivateRepoSSHAgent(t)
+		socket = c.Host().UnixSocket(sockPath)
+	}
+
+	token := ""
+	if tc.token != "" {
+		decoded, err := base64.StdEncoding.DecodeString(tc.token)
+		require.NoError(t, err)
+		token = strings.TrimSpace(string(decoded))
+	}
 
 	return func(ctr *dagger.Container) *dagger.Container {
-		sock := c.Host().UnixSocket(sockPath)
-		if sock != nil {
-			// Ensure that HOME env var is set, to ensure homePath expension in test suite
-			homeDir, _ := os.UserHomeDir()
-			ctr = ctr.WithEnvVariable("HOME", homeDir)
```

Comment:
I'm guessing from here https://github.com/dagger/dagger/pull/8365#discussion_r1762068678, that this isn't required? `SSH_AUTH_SOCK` doesn't contain `~`.

Additionally this seems to make these tests not work when running them as a normal user (e.g. using `./hack/with-dev go test ...`)